umbra/umbra_64.c

```diff
@@ -1261,6 +1262,34 @@ umbra_insert_app_to_shadow_arch(void *drcontext,
                                 reg_id_t *scratch_regs,
                                 int num_scratch_regs)
 {
+#if defined(AARCH64)
+    if (num_scratch_regs >= umbra_num_scratch_regs_for_translation_arch())
```

Comment:
This is now incorrect :) We need to return DRMF_ERROR_INVALID_PARAMETER if num_scratch_regs is less than the required number of scratch registers as determined via umbra_num_scratch_regs_for_translation_arch().

Recall `reg_id_t *scratch_regs` and `int num_scratch_regs` are user controlled parameters. For umbra_insert_app_to_shadow_arch() to work, the user needs to provide a sufficient number of scratch registers. It is for this reason why we are including the check.

Reply:
yep, wasn't really thinking evidently, but woke up this morning and corrected (before reading this :P)

Reply:
Awesome! Thanks for the quick fix. We just have the last pending issue regarding indentation, and then we are good for merging.